### PR TITLE
First cut

### PR DESCRIPTION
**Pull request by CapnKernel**
There'll be another drop later that fixes a few things, adds a backend, and makes it conform better to the spec.

### CODE REVIEW

**Comment by sesh** [comment]
OMG UNDER CONSTRUCTION GIF ALERT!

:truck: :articulated_lorry: :thumbsup:

File: guestbook/models.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from django.db import models
+
+class Comment(models.Model):
+    name = models.CharField(max_length=255)
+    email_address = models.EmailField(max_length=254, blank=True, help_text="optional")
```
Comment (sesh) [inline guestbook/models.py:5]: `EmailField` was updated in 1.8 to use `max_length=254` by default.

File: guestbook/templates/mypage.html [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            <td>{{ comment.comment }}</td>
+          </tr>
+        {% endfor %}
+      </table>
+        
+      <p>{{ comments_count }} {{ comments_count|pluralize:"entry,entries" }}.</p>
+        
+      <h3>Tell us what you think!</h3>
+    {% else %}
+      <h3>Be the first to leave a comment!</h3>
```
Comment (sesh) [inline guestbook/templates/mypage.html:31]: Nice work handling the no comments situation!

File: guestbook/templates/base.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    {% if messages %}
+      <h3>Messages</h3>
+      <ul>
+        {% for message in messages %}
+          <li>{{ message|safe }}</li>
+        {% endfor %}
+      </ul>
+    {% endif %}
+
+    {% block content %}<p>This space for rent.</p>{% endblock %}
```
Comment (sesh) [inline guestbook/templates/base.html:16]: Classic.

**Comment by sesh** [comment]
Hey @CapnKernel nice work.

Made a couple of quick notes, will give you some proper feedback on Monday!

File: guestbook/templates/base.html [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    <hr>
+    <img src="{% static 'img/agbdunderconstruction7.gif' %}" />
```
Comment (MarkusH) [inline guestbook/templates/base.html:19]: :+1: :heart:

File: guestbook/templates/mypage.html [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        {% for comment in comments %}
+          <tr>
+            <td>{{ comment.name }}</td>
+            <td>{{ comment.created }} ({{ comment.created|timesince }} ago)</td>
+            <td>{{ comment.comment }}</td>
+          </tr>
+        {% endfor %}
+      </table>
+        
+      <p>{{ comments_count }} {{ comments_count|pluralize:"entry,entries" }}.</p>
```
Comment (MarkusH) [inline guestbook/templates/mypage.html:27]: Nice!

File: guestbook/templates/mypage.html [hunk 6]
```diff
@@ -0,0 +1,41 @@
+{% extends "base.html" %}
+
+{% block title %}My page{% endblock %}
+
+{% block content %}
+  <h1>My page</h1>
+
+  {% with comments_count=comments.count %}
```
Comment (MarkusH) [inline guestbook/templates/mypage.html:8]: Good thinking to not execute the same database query (count public comments) multiple times!

File: guestbook/views.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
+from django import forms
+from django.contrib import messages
+from django.shortcuts import render
+from django.core.urlresolvers import reverse
+from django.http import HttpResponseRedirect
+from django.views.decorators.http import require_http_methods
+
+from .models import Comment
+
+class CommentForm(forms.ModelForm):
```
Comment (MarkusH) [inline guestbook/views.py:10]: You would normally move this form into `forms.py`, even if that's the only form in that app. It keeps things clear and well structured.

File: guestbook/views.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+@require_http_methods(('GET', 'POST'))
+def mypage(request):
+    if request.method == 'POST':
+        form = CommentForm(request.POST)
+        if form.is_valid():
+            form.save()
+            messages.info(request, 'Thank you for your comment.')
+
+            return HttpResponseRedirect(reverse('mypage'))
```
Comment (MarkusH) [inline guestbook/views.py:23]: there's a function called `redirect` in `django.shortcuts` that directly takes a URL name: `return redirect('mypage')`

**Comment by MarkusH** [comment]
Looks good so far :smile: .

A few more general notes:
1. I like that you experiment with (unknown?) features, such as `django.contrib.messages`
2. I don't see a usage of the :meat_on_bone: substrings (https://twitter.com/m_holtermann/status/634619186468098048)

File: guestbook/templates/base.html (hunk 4 above)
Comment (sesh) [inline guestbook/templates/base.html:19]: :articulated_lorry: :heart:

File: assignment1/tests.py [hunk 9]
```diff
@@ -0,0 +1,8 @@
+from django.test import TestCase, Client
+
+class LessonOneTests(TestCase):
+
+    def test_hello_melbdjango(self):
```
Comment (funkybob) [inline assignment1/tests.py:5]: Nice... even a simple smoke test is better than no tests at all :+1:

File: guestbook/urls.py [hunk 10]
```diff
@@ -0,0 +1,9 @@
+from django.conf.urls import include, url
+from django.http import HttpResponseRedirect
+from django.core.urlresolvers import reverse
+
+urlpatterns = [
+    url(r'^$', lambda request: HttpResponseRedirect(reverse('mypage'))),
```
Comment (funkybob) [inline guestbook/urls.py:6]: Cute trick... you could also have used the [`redirect`](https://docs.djangoproject.com/en/1.8/topics/http/shortcuts/#redirect) shortcut:

```
from django.shortcuts import redirect

...

url(r'^$', lambda request: return redirect('mypage')),
```

There is also the [`RedirectView`](https://docs.djangoproject.com/en/1.8/ref/class-based-views/base/#django.views.generic.base.RedirectView):

```
from django.views.generic import RedirectView

url(r'^$', RedirectView.as_view(pattern_name='mypage')),
```

**Pull request closed without merging**